common/lib/xmodule/xmodule/modulestore/tests/test_mixed_modulestore.py

```diff
@@ -506,6 +512,213 @@ def test_has_changes(self, default_ms):
         component = self.store.publish(component.location, self.user_id)
         self.assertFalse(self.store.has_changes(component))
 
+    @ddt.data(ModuleStoreEnum.Type.mongo, ModuleStoreEnum.Type.split)
+    def test_draft_has_changes_before_export_and_after_import(self, default_ms):
+        """
+        Tests that an unpublished unit remains with no changes across export and re-import.
+        """
+        with MongoContentstoreBuilder().build() as contentstore:
+            # initialize the mixed modulestore
+            self._initialize_mixed(contentstore=contentstore, mappings={})
+
+            with self.store.default_store(default_ms):
+
+                source_course_key = self.store.make_course_key("org.source", "course.source", "run.source")
+                self._create_course(source_course_key)
+
+                # Create a dummy component to test against and don't publish it.
+                draft_xblock = self.store.create_item(
+                    self.user_id,
+                    self.course.id,
+                    'vertical',
+                    block_id='test_vertical'
+                )
+                # Not yet published, so changes are present
+                self.assertTrue(self._has_changes(draft_xblock.location))
+
+                export_course_to_xml(
+                    self.store,
+                    contentstore,
+                    source_course_key,
+                    self.export_dir,
+                    'exported_source_course',
+                )
+
+                import_course_from_xml(
+                    self.store,
+                    'test_user',
+                    self.export_dir,
+                    source_dirs=['exported_source_course'],
+                    static_content_store=contentstore,
+                    target_id=source_course_key,
+                    create_if_not_present=True,
+                    raise_on_failure=True,
+                )
+
+                # Verify that the imported block still is a draft, i.e. has changes.
+                self.assertTrue(self._has_changes(draft_xblock.location))
```

Comment:
Optional: It might be worth checking the contents of the block fields, as well, to verify that those haven't changed.